fendermint/app/src/app.rs

```diff
@@ -672,6 +652,16 @@ where
         let db = self.state_store_clone();
         let state = self.committed_state()?;
         let mut state_params = state.state_params.clone();
+
+        // Notify the snapshotter. We don't do this in `commit` because *this* is the height at which
+        // this state has been officially associated with the application hash, which is something
+        // we will receive in `offer_snapshot` and we can compare. If we did it in `commit` we'd
```

Comment:
This is done to ensure that the `state_params` that we are requesting belongs to the right height because in `end_block` we haven't yet computed the up to date `state_params`, right?